src/main/java/fi/aalto/cs/apluscourses/ui/ideactivities/ComponentLocator.java

```diff
@@ -21,19 +21,20 @@ public static boolean hasChildOfClass(@NotNull Container c, @NotNull String clas
     return !getComponents(c, comp -> isComponentOfClass(comp, classSubstring)).isEmpty();
   }
 
+  /**
+   * Determines whether a component is related to an action of a specific class.
+   *
+   * @param c Component
+   * @param classSubstring Part of the action class name.
+   * @return True, if component matches to the action, otherwise false.
+   */
   public static boolean hasActionOfClass(@NotNull JButton c, @NotNull String classSubstring) {
     var action = c.getAction();
     var actionProperty = c.getClientProperty(CustomComponentAction.ACTION_KEY);
 
-    if (action != null && action.getClass().getName().contains(classSubstring)) {
-      return true;
-    }
-
-    if (actionProperty != null && actionProperty.getClass().getName().contains(classSubstring)) {
-      return true;
-    }
+    return action != null && action.getClass().getName().contains(classSubstring)
+        || actionProperty != null && actionProperty.getClass().getName().contains(classSubstring);
```

Comment:
I suggest adding parentheses like that:
`return (action != null && action.getClass().getName().contains(classSubstring)) || (actionProperty != null && actionProperty.getClass().getName().contains(classSubstring));`

Reply:
Checkstyle (or some other linter, can't remember) wants operators like `||` to be in the beginning of the new line instead of ending the previous line. About parentheses: they are sometimes useful but in this case, I think the expression is readable enough without them. Symmetry between the lines helps here.